httphandler/storage/apiserver.go

```diff
@@ -0,0 +1,588 @@
+			return updateErr
+		})
+		if retryErr != nil {
+			logger.L().Ctx(ctx).Warning("failed to update WorkloadConfigurationScanSummary manifest in storage", helpers.Error(err),
+				helpers.String("name", manifest.Name))
+		} else {
+			logger.L().Debug("updated WorkloadConfigurationScanSummary manifest in storage",
+				helpers.String("name", manifest.Name))
+		}
+	case err != nil:
+		logger.L().Ctx(ctx).Warning("failed to store WorkloadConfigurationScanSummary manifest in storage", helpers.Error(err),
+			helpers.String("name", manifest.Name))
+		return nil, err
+	default:
+		logger.L().Debug("stored WorkloadConfigurationScanSummary manifest in storage",
+			helpers.String("name", manifest.Name))
+	}
+	return &manifest, nil
+}
+
+func getManifestObjectLabelsAndAnnotations(ctx context.Context, resource workloadinterface.IMetadata, relatedObjects []workloadinterface.IMetadata) (labels map[string]string, annotations map[string]string, err error) {
+	m := make(map[string]string)
+	if len(relatedObjects) > 0 {
+		m[v1.RbacResourceMetadataKey] = "true"
+
+		for i := range relatedObjects {
+			relatedObject := relatedObjects[i]
+			switch relatedObject.GetKind() {
+			case "Role":
+				m[v1.RoleNameMetadataKey] = relatedObject.GetName()
+				m[v1.RoleNamespaceMetadataKey] = relatedObject.GetNamespace()
+			case "RoleBinding":
+				m[v1.RoleBindingNameMetadataKey] = relatedObject.GetName()
+				m[v1.RoleBindingNamespaceMetadataKey] = relatedObject.GetNamespace()
+			case "ClusterRole":
+				m[v1.ClusterRoleNameMetadataKey] = relatedObject.GetName()
+			case "ClusterRoleBinding":
+				m[v1.ClusterRoleBindingNameMetadataKey] = relatedObject.GetName()
+			default:
+				return nil, nil, fmt.Errorf("unknown related object kind %s", relatedObject.GetKind())
+			}
+		}
+	}
+
+	m[v1.ApiGroupMetadataKey], m[v1.ApiVersionMetadataKey] = k8sinterface.SplitApiVersion(resource.GetApiVersion())
+	m[v1.KindMetadataKey] = resource.GetKind()
+	m[v1.NameMetadataKey] = resource.GetName()
+	if k8sinterface.IsResourceInNamespaceScope(resource.GetKind()) {
+		m[v1.NamespaceMetadataKey] = resource.GetNamespace()
+	}
+
+	annotations = m
+	labels = make(map[string]string)
+	maps.Copy(labels, m)
+	names.SanitizeLabelValues(labels)
+
+	return labels, annotations, nil
+}
+
+func getRelatedObjects(resource *reporthandling.Resource) []workloadinterface.IMetadata {
+	obj := resource.GetObject()
+
+	if objectsenvelopes.IsTypeRegoResponseVector(obj) {
+		regoResponseVectorObject := objectsenvelopes.NewRegoResponseVectorObject(obj)
+		return regoResponseVectorObject.GetRelatedObjects()
+	}
+
+	return []workloadinterface.IMetadata{}
+}
+
+func GetWorkloadScanK8sResourceName(ctx context.Context, resource workloadinterface.IMetadata, relatedObjects []workloadinterface.IMetadata) string {
+	var sb strings.Builder
+
+	// apiVersion (optional for rego response vector)
+	apiVersion := resource.GetApiVersion()
```

Comment:
need to understand why in some cases this might be empty - it might cause an issue (to check)